fiftyone/utils/youtube.py

```diff
@@ -0,0 +1,461 @@
+"""
+Utilities for working with `YouTube <https://youtube.com>`.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import itertools
+import multiprocessing
+import multiprocessing.dummy
+import os
+
+import numpy as np
+
+import eta.core.utils as etau
+import eta.core.video as etav
+
+import fiftyone.core.utils as fou
+
+pytube = fou.lazy_import(
+    "pytube", callback=lambda: fou.ensure_package("pytube"),
+)
+
+
+logger = multiprocessing.get_logger()
+
+
+def download_youtube_videos(
+    urls,
+    download_dir=None,
+    video_paths=None,
+    clip_segments=None,
+    ext=".mp4",
+    only_progressive=True,
+    resolution="highest",
```

Comment:
I read the pytube docs and found that `ext=".mp4", only_progressive=True, resolution="highest"` should generally result in a 720p video being downloaded. This makes sense as a default behavior I think.

Certain videos from popular creators might have crazy 4k streams available (stored as non-progressive, non-MP4 streams, per my research), which are generally not useful for ML use cases. However, the user can still pass `ext=None, only_progressive=False` to get their hands on the absolute highest quality stream.